library/Zend/Stdlib/ArrayUtils.php

```diff
@@ -238,12 +238,7 @@ public static function iteratorToArray($iterator, $recursive = true)
                 continue;
             }
 
-            if ($value instanceof Traversable) {
-                $array[$key] = static::iteratorToArray($value, $recursive);
-                continue;
-            }
-
-            if (is_array($value)) {
+            if (is_array($value) || $value instanceof Traversable) {
                 $array[$key] = static::iteratorToArray($value, $recursive);
```

Comment:
Wondering if we can optimize this loop by iterating `if (is_array())` (instead of recursing if it is an `array`)